### PR TITLE
Add REST API endpoints for trashing/untrashing documents

### PR DESCRIPTION
**Pull request by buchi**

### CODE REVIEW

File: opengever/api/trash.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+import plone.protect.interfaces
+
+
+class Trash(Service):
+    """Trash a document"""
+
+    def reply(self):
+
+        # Disable CSRF protection
+        if 'IDisableCSRFProtection' in dir(plone.protect.interfaces):
```
Comment (lukasgraf) [inline opengever/api/trash.py]: That check isn't necessary here - we have a [hard dependency on `plone.protect >= 3.0.15`](https://github.com/4teamwork/opengever.core/blob/master/setup.py#L139) in GEVER, that request layer will always be importable (same for `Untrash` below).

File: opengever/api/configure.zcml [hunk 2]
```diff
@@ -35,6 +36,22 @@
       permission="opengever.document.Cancel"
       />
 
+  <plone:service
+      method="POST"
+      name="@trash"
+      for="opengever.document.document.IDocumentSchema"
```
Comment (lukasgraf) [inline opengever/api/configure.zcml]: If mails (`ftw.mail.mail`) should also be trashable (which they are via the UI), this should be registered for `opengever.document.behaviors.IBaseDocument`.

**Comment by lukasgraf** [comment]
Could you please also add a short mention of this endpoint to the [GEVER API docs](https://docs.onegovgever.ch/dev-manual/api/)? (`docs/public/dev-manual/api/` - maybe similar to the already documented `@scanin` endpoint).

File: opengever/api/trash.py (hunk 1 above)
Comment (buchi) [reply opengever/api/trash.py]: check removed

File: opengever/api/configure.zcml (hunk 2 above)
Comment (buchi) [reply opengever/api/configure.zcml]: changed to `opengever.document.behaviors.IBaseDocument`

**Comment by buchi** [comment]
Moved permission and condition checks from browser view to trasher adapter.
Added some short docs.

**Pull request merged**